storage-proto/src/lib.rs

```diff
@@ -109,6 +109,22 @@ impl From<UiTokenAmount> for StoredTokenAmount {
     }
 }
 
+struct StoredTransactionError(Vec<u8>);
+
+impl From<StoredTransactionError> for TransactionError {
+    fn from(value: StoredTransactionError) -> Self {
+        let bytes = value.0;
+        bincode::deserialize::<Self>(&bytes).expect("transaction error to deserialize from bytes")
```

Comment:
nit: is the `Self` needed here? I think the compiler should be able to figure it out
```suggestion
        bincode::deserialize(&bytes).expect("transaction error to deserialize from bytes")
```